Sources/Noora/Noora.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import Mockable
+
+@Mockable
+public protocol Noorable {
+    func singleChoicePrompt<T: CaseIterable & CustomStringConvertible & Equatable>(
+        question: String,
+        options: T.Type
+    )
+
+    func singleChoicePrompt<T: CaseIterable & CustomStringConvertible & Equatable>(
+        title: String?,
+        question: String,
+        description: String?,
+        options: T.Type,
+        collapseOnSelection: Bool,
+        theme: NooraTheme,
+        terminal: Terminal
+    )
+
+    func yesOrNoChoicePrompt(
+        title: String?,
+        question: String
+    )
+
+    func yesOrNoChoicePrompt(
+        title: String?,
+        question: String,
+        defaultAnswer: Bool,
+        description: String?,
+        collapseOnSelection: Bool,
+        theme: NooraTheme,
+        terminal: Terminal
+    )
+}
+
+public struct Noora {
+    public init() {}
```

Comment:
If we're not storing any state here, the `singleChoicePrompt` can be a static method, so you can run `Noora.singleChoicePrompt(...)` instead of `Noora().singleChoicePrompt()`. Alternatively, the `Noora` struct can hold the theme as I assume that will be the same across runs.

Reply:
I made them instance methods to facilitate testing to the users of the API through dependency injection. I'll make the theme a property for the reasons you mentioned.